dali/operators/util/normal_distribution_op.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/util/normal_distribution_op.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+
+DALI_SCHEMA(NormalDistribution)
+                .DocStr(R"code(Creates a tensor is data distributed normally.
+The shape of the provided tensor is implied by the tensor provided as an input.
```

Comment:
Mhh. I would make it optional as an AdditionalInput maybe? I can imagine that we want to define the shape on our own and let the arithm ops to extend it accordingly.

Reply:
Do I understood correctly, that you'd like to have sth like:
```
NormalDistribution(mean=, stddev=, shape= )
```
And when `shape` is not provided, it's deducted from the input?

The problem would appear for `ArgumentInput` in this case, since we need shape per every tensor in batch. Then we bump into the `ArgumentInput` on a GPU and the need of copying it back and forth